### ISSUE TITLE
`@AnnotatedFor` in a stub file does not enable warnings under `-AonlyAnnotatedFor` or `-AuseConservativeDefaultsForUncheckedCode=source`

### ISSUE THREAD

**Issue by aosen-xiong** [eisop/checker-framework#1983]
## Reproducer

`SuppressionTest.java`:

```java
package annotatedforsuppression;

public class SuppressionTest {
    Object f = null;
}
```

`SuppressionTest.astub`:

```java
package annotatedforsuppression;

import org.checkerframework.framework.qual.AnnotatedFor;

@AnnotatedFor("nullness")
public class SuppressionTest {
}
```

```
javac -processor org.checkerframework.checker.nullness.NullnessChecker \
      -AonlyAnnotatedFor -Astubs=SuppressionTest.astub SuppressionTest.java
```

## Expected

`SuppressionTest.java:4: error: (assignment.type.incompatible)`

The stub puts the class in the scope of `@AnnotatedFor("nullness")`, and `-AonlyAnnotatedFor` suppresses warnings only *outside* such a scope.

## Actual

No error. The warning is suppressed as if the class had no `@AnnotatedFor`.

Same result with `-AuseConservativeDefaultsForUncheckedCode=source` in place of `-AonlyAnnotatedFor`. Without `-Astubs`, suppression is correct in both cases.

**Comment by aosen-xiong** [comment]
`SourceChecker.isAnnotatedForThisCheckerOrUpstreamChecker` uses `elt.getAnnotation`, which doesn't see stub files, while `QualifierDefaults.isElementAnnotatedForThisChecker` uses `getDeclAnnotation`, which does.

So, refactoring in #1331 fixes this issue.

**Comment by aosen-xiong** [comment]
Correction: the expected behavior in this report is wrong. Without -AmergeStubsWithSource, a stub for a class that is being compiled is ignored, so suppressing the warning is correct.
